distribution_metadata.c

```diff
@@ -389,7 +391,7 @@ IsDistributedTable(Oid tableId)
 	 * The query below hits the partition metadata table, so if we don't detect
 	 * that and short-circuit, we'll get infinite recursion in the planner.
 	 */
-	if (tableId == partitionMetadataTableOid)
+	if (IsSystemNamespace(tableNamespaceOid) || tableId == partitionMetadataTableOid)
```

Comment:
I see, so there are two use-cases which hits the recursion use-case: queries against the CitusDB partition table as well as queries against the pg_shard partition table? The clause captures the CitusDB case, and the second captures the pg_shard case?

If so, could we modify the comment to make that clear?